components/event-bus/before-commit.sh

```diff
@@ -0,0 +1,126 @@
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+INVERTED='\033[7m'
+NC='\033[0m' # No Color
+
+echo -e "${INVERTED}"
+echo "USER: " + $USER
+echo "PATH: " + $PATH
+echo "GOPATH:" + $GOPATH
+echo -e "${NC}"
+
+##
+# DEP ENSURE
+##
+dep ensure -v --vendor-only
+ensureResult=$?
+if [ ${ensureResult} != 0 ]; then
+	echo -e "${RED}✗ dep ensure -v --vendor-only${NC}\n$ensureResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ dep ensure -v --vendor-only${NC}"
+fi
+
+##
+# GO BUILD
+##
+buildEnv=""
+if [ "$1" == "$CI_FLAG" ]; then
+	# build binary statically
+	buildEnv="env CGO_ENABLED=0"
+fi
+
+${buildEnv} go build -v -a -installsuffix cgo -o ./cmd/event-bus-publish/event-bus-publish ./cmd/event-bus-publish/
+${buildEnv} go build -v -a -installsuffix cgo -o ./cmd/event-bus-push/event-bus-push ./cmd/event-bus-push/
+${buildEnv} go build -v -a -installsuffix cgo -o ./cmd/event-bus-sv/event-bus-sv ./cmd/event-bus-sv/
+
+goBuildResult=$?
+if [ ${goBuildResult} != 0 ]; then
+	echo -e "${RED}✗ go build${NC}\n$goBuildResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ go build${NC}"
+fi
+
+##
+# DEP STATUS
+##
+echo "? dep status"
+depResult=$(dep status -v)
+if [ $? != 0 ]; then
+	echo -e "${RED}✗ dep status\n$depResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ dep status${NC}"
+fi
+
+##
+# GO TEST
+##
+echo "? go test"
+go test ./...
+# Check if tests passed
+if [ $? != 0 ]; then
+	echo -e "${RED}✗ go test\n${NC}"
+	exit 1
+else echo -e "${GREEN}√ go test${NC}"
+fi
+
+goFilesToCheck=$(find . -type f -name "*.go" | egrep -v "\/vendor\/|_*/automock/|_*/testdata/|/pkg\/|_*export_test.go")
+
+##
+#  GO LINT
+##
+go build -o golint-vendored ./vendor/github.com/golang/lint/golint
+buildLintResult=$?
+if [ ${buildLintResult} != 0 ]; then
+	echo -e "${RED}✗ go build lint${NC}\n$buildLintResult${NC}"
+	exit 1
+fi
+
+golintResult=$(echo "${goFilesToCheck}" | xargs -L1 ./golint-vendored)
+rm golint-vendored
+
+if [ $(echo ${#golintResult}) != 0 ]; then
+	echo -e "${RED}✗ golint\n$golintResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ golint${NC}"
+fi
+
+##
+# GO IMPORTS & FMT
+##
+go build -o goimports-vendored ./vendor/golang.org/x/tools/cmd/goimports
+buildGoImportResult=$?
+if [ ${buildGoImportResult} != 0 ]; then
+	echo -e "${RED}✗ go build goimports${NC}\n$buildGoImportResult${NC}"
+	exit 1
+fi
+
+goImportsResult=$(echo "${goFilesToCheck}" | xargs -L1 ./goimports-vendored -w -l)
+rm goimports-vendored
+
+if [ $(echo ${#goImportsResult}) != 0 ]; then
+	echo -e "${RED}✗ goimports and fmt ${NC}\n$goImportsResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ goimports and fmt ${NC}"
+fi
+
+##
+# GO VET
+##
+packagesToVet=("./api/..." "./cmd/..." "./internal/..." "./test/...")
+
+for vPackage in "${packagesToVet[@]}"; do
+	vetResult=$(go vet ${vPackage})
+	if [ $(echo ${#vetResult}) != 0 ]; then
+		echo -e "${RED}✗ go vet ${vPackage} ${NC}\n$vetResult${NC}"
+		exit 1
+	else echo -e "${GREEN}√ go vet ${vPackage} ${NC}"
+	fi
+done
+
+rm ./cmd/event-bus-publish/event-bus-publish
```

Comment:
the before-commit.sh might go away in the future. We should not have any customizations in it.

Reply:
First of all, The pre-commit file is used by many kyma components, and each component customizes it to meet their needs.

Secondly, For event-bus this is totally a new file, so Why can't we customize it to meet our needs? Even if it will go away in the future, but still it will be working for sometime, so Why not make it work for event-bus in the meantime.

Finally, See lines 36, 37 and 38 those are pretty custom lines specifically to compile and build event-bus apps. Without these lines the pre-commit phase is useless. The same goes for lines 124, 125 and 126 to clean-up the apps' binaries.